cloud-nio/cloud-nio-impl-drs/src/test/scala/cloud/nio/impl/drs/DrsCloudNioFileProviderSpec.scala

```diff
@@ -39,6 +39,8 @@ class DrsCloudNioFileProviderSpec extends AnyFlatSpecLike with Matchers with Moc
     fileSystemProvider.getScheme should be("drs")
     fileSystemProvider.getHost("drs://dg.123/abc") should be("dg.123")
     fileSystemProvider.getHost("drs://dg.example.com/abc") should be("dg.example.com")
+    fileSystemProvider.getHost("drs://dg.4503:dg.4503/abc") should be("dg.4503")
+    fileSystemProvider.getHost("drs://dg.4503:abc") should be("dg.4503")
```

Comment:
```suggestion
    fileSystemProvider.getHost("drs://dg.4503:dg.4503/abc") should be("dg.4503")
    fileSystemProvider.getHost("drs://dg.4503:abc") should be("dg.4503")
    fileSystemProvider.getHost("drs://:abc") should be(<fail gracefully>)
    fileSystemProvider.getHost("drs://dg.4503:") should be(<fail gracefully>)
```

Reply:
The `getHost` function, before these changes, would return an empty string if it couldn't find a valid host, so I've retained that behavior.  Does that count as failing gracefully?  If not, what benefit would be gained by throwing an error?

Reply:
Yes, that counts as failing gracefully.

Ungraceful would be if it threw a random exception.

Reply:
[TOL] Although it would be pretty cool to use `Option` and return `None` in that case

Reply:
Added the two tests, although `drs://dg.4503:` returns `dg.4503` as host, because that URI has a host, just no port or filepath.